src/NosCore.Database/DAL/ItemInstanceDAO.cs

```diff
@@ -35,250 +38,267 @@
 
 namespace NosCore.Database
 {
-    public class ItemInstanceDao : IGenericDao<IItemInstanceDto>
+    public class ItemInstanceDao : Dao<ItemInstance, IItemInstanceDto, Guid>
     {
         private readonly ILogger _logger;
-        private readonly PropertyInfo? _primaryKey;
+        private readonly PropertyInfo[] _primaryKey;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unread private field '_primaryKey' or refactor the code to use its value.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5326532)